### DISCUSSION TITLE
fyi: using ebusd with mqtt and node-red to log to volkszaehler

### DISCUSSION THREAD

**Discussion by andig** [john30/ebusd#718] [Show and tell]
Hi everyone,

I've pieced together a small node-red flow that:

  - pulls ebusd via mqtt
  - converts the results (especially Status01 as well as pump and vent state) into numbers
  -  pushes the results- if updated- to volkszaehler

Any interest please let me know to add some details here.

**Comment by NemoN** [comment]
> Any interest please let me know to add some details here.

please provide details

**Comment by andig** [comment]
<img width="1152" alt="screen shot 2017-03-17 at 19 13 37" src="https://cloud.githubusercontent.com/assets/184815/24057143/a5be92e2-0b46-11e7-8c39-7a6e97fe21c9.png">
<img width="1152" alt="screen shot 2017-03-17 at 19 14 18" src="https://cloud.githubusercontent.com/assets/184815/24057145/a84bd0f6-0b46-11e7-887e-054c10a02d17.png">
<img width="1152" alt="screen shot 2017-03-17 at 19 15 02" src="https://cloud.githubusercontent.com/assets/184815/24057151/aac7574c-0b46-11e7-9a4f-efb04be546bc.png">

**Comment by andig** [comment]
I can share the flows if you want

**Comment by NemoN** [comment]
please share the flows :-)

**Comment by andig** [comment]
> please share the flows

Any questions/improvements please let me know. Error handling is rudimentary since I obviously know that the required channels exist ;)

https://gist.github.com/andig/9b1d715ba1fcb60b6fb646bf69dfb7f3

**Comment by NemoN** [comment]
@andig do you have a screenshot from the final charts in VZ?

**Comment by andig** [comment]
@NemoN Etwa so in der Monatsansicht, nur die interessantesten Kanäle ausgewählt:

![unbenannt](https://cloud.githubusercontent.com/assets/184815/25275163/2ec1fc7e-2694-11e7-9770-25d8f437698b.png)

**Comment by john30** [comment]
this can be achieved easier now by using the new ebusd argument --mqttjson since 259f9b76ef44b4d5873443e24d911a604fe271f4

**Comment by andig** [comment]
@john30 couldn't find it in the wiki and don't want to break my prod setup. Could you add json example to the docs?

**Comment by john30** [comment]
its just using "--mqttjson" additionally and then you'll get the data in json format just as by retrieving from HTTP port /data

**Comment by andig** [comment]
Sure, but what do you mean by "json format"? I'm getting a single value per channel only, what advantage does json have here?

**Comment by john30** [comment]
then leave it as it is :-)
